cmd/controllers/main.go

```diff
@@ -290,13 +301,20 @@ func main() {
 }
 
 // TODO: if the argument list will grow, we should create a ControllerSetupOptions struct instead.
-func setupControllers(mgr ctrl.Manager, runtimeConfig config.RuntimeConfig, certsReady chan struct{}, disableWebhook bool) {
+func setupControllers(mgr ctrl.Manager, runtimeConfig config.RuntimeConfig, certsReady chan struct{}) {
 	// The controllers won't work until the webhooks are operating,
 	// and the webhook won't work until the certs are all in places.
-	if !disableWebhook {
+	if certsReady == nil {
+		setupLog.Info("webhook setup skipped due to --disable-webhook flag")
+	} else {
 		setupLog.Info("waiting for the cert generation to complete")
 		<-certsReady
 		setupLog.Info("certs ready")
+
+		if err := apiwebhook.SetupBackendRuntimeWebhook(mgr); err != nil {
```

Comment:
Looking at the code, I don't think the order will have any particular effect.

Let me know if I'm overlooking something.

Reply:
Let's change the name of SetupBackendRuntimeWebhook to SetupModelAdapterWebhook, it will setup the webhook.